pkg/bisect/bisect.go

```diff
@@ -50,12 +50,14 @@ type ReproConfig struct {
 	C    []byte
 }
 
+// TODO: use better interface names?
 type env struct {
 	cfg       *Config
 	repo      vcs.Repo
 	bisecter  vcs.Bisecter
+	commitEnv vcs.EnvForCommitOverridable
```

Comment:
This looks quite hacky. EnvForCommit is still part of Bisecter interface, but we don't use EnvForCommit from bisecter. So at the very least we need to remove EnvForCommit from Bisecter entirely. But it looks better to mock the whole Bisecter implementation and don't use the Linux one.
The Linux bisecter tries to run "get_maintainers.pl", which will fail as well, tries to do cherry-picks that will fail in our test, etc.
The only part of Linux Bisecter that we could test and want to test is previousReleaseTags. We could move most of it into pkg/vcs.git object and then create own test Bisecter implementation on top of pkg/vcs.git. This looks like much cleaner solution to me.

Reply:
I see, thanks, I've made the change and uploaded a PR.